components/fxa-client/src/oauth.rs

```diff
@@ -66,7 +68,12 @@ impl FirefoxAccount {
         let since_epoch = SystemTime::now()
             .duration_since(UNIX_EPOCH)
             .map_err(|_| ErrorKind::IllegalState("Current date before Unix Epoch."))?;
-        let expires_at = since_epoch.as_secs() + resp.expires_in;
+        let expires_in = if 0 < ttl && ttl < resp.expires_in {
+            ttl
+        } else {
+            resp.expires_in
+        };
+        let expires_at = since_epoch.as_secs() + expires_in;
```

Comment:
Shouldn't we pass the TTL to the server in the request parameters instead? cc @rfk

Reply:
Yes we should. I expect this would need to be passed in to `access_token_with_refresh_token` and `access_token_with_session_token` in order to pass it on to the server. We should then rely on the server's calculation of `expires_in` rather than doing our own calculation here (since the server is free to ignore the requested ttl and apply e.g. its own bounds on how big or small the ttl is allowed to be).